### PR TITLE
refactor(test): decouple full sync from other tests

### LINKED ISSUE #3712: CI fails because the API rate limit is exceeded
## Motivation

CI has failed multiple times over the last few days with:
> Error: API rate limit exceeded for installation ID 1523599.

https://github.com/ZcashFoundation/zebra/runs/5399888051?check_suite_focus=true#step:5:69

### PR DESCRIPTION
**Pull request by gustavovalverde**
## Motivation
- We are hitting GitHub API limits, blocking merges and reviews.
- We're also having actions being run without being actually needed

Closes https://github.com/ZcashFoundation/zebra/pull/3733
Fixes: https://github.com/ZcashFoundation/zebra/issues/3712
Depends-On: #3728

## Solution

- As the full sync requires to be run just once and isolated, we're running this test in a separate workflow, after a PR has been approved.
- Some actions got improved conditions to run just when needed

## Review
@dconolly and @teor2345  can review this

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -346,9 +347,9 @@ jobs:
       # Get the latest uploaded txt with the disk SHORT_SHA from this workflow
       - name: Download latest disk state SHORT_SHA
         uses: dawidd6/action-download-artifact@v2.17.0
-        # Just search for the latest uploaded artifact if the previous job was skipped,
+        # Just search for the latest uploaded artifact if the previous disk regeneration job was skipped,
         # otherwise use the output from ${{ needs.regenerate-stateful-disks.outputs.disk_short_sha }}
-        if: ${{ needs.regenerate-stateful-disks.steps.create-instance.outcome == 'skipped' }}
+        if: ${{ needs.regenerate-stateful-disks.outputs.any_changed != 'true' ||  || github.event.inputs.regenerate-disks != 'true'}}
```
Comment (dconnolly) [inline .github/workflows/test.yml]: Do you need two sets of `||` here?
image

Comment (gustavovalverde) [reply .github/workflows/test.yml]: Oops! [bcfea78](https://github.com/ZcashFoundation/zebra/pull/3735/commits/bcfea783662fff10c3d2c81b2013749195c4eebd)

File: .github/workflows/test-full-sync.yml [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+          --command="docker logs --follow ${{ env.ZEBRA_CONTAINER }}"
+        env:
+          ZEBRA_CONTAINER: ${{ steps.get-container-name.outputs.zebra_container }}
+
+      - name: Delete test instance
+        # Do not delete the instance if the sync timeouts in GitHub
+        if: ${{ steps.sync-past-checkpoint.outcome == 'success' || steps.sync-past-checkpoint.outcome == 'failure' }}
+        continue-on-error: true
+        run: |
+          gcloud compute instances delete "sync-tests-${{ env.GITHUB_HEAD_REF_SLUG_URL || env.GITHUB_REF_SLUG_URL }}-${{ env.GITHUB_EVENT_PULL_REQUEST_HEAD_SHA_SHORT || env.GITHUB_SHA_SHORT }}" --delete-disks all --zone "${{ env.ZONE }}"
```
Comment (dconnolly) [inline .github/workflows/test-full-sync.yml]:
```suggestion
      - name: Full sync mainnet
        id: full-sync-mainnet
        run: |
          gcloud compute ssh \
          sync-tests-${{ env.GITHUB_HEAD_REF_SLUG_URL || env.GITHUB_REF_SLUG_URL }}-${{ env.GITHUB_EVENT_PULL_REQUEST_HEAD_SHA_SHORT || env.GITHUB_SHA_SHORT }} \
          --zone ${{ env.ZONE }} \
          --quiet \
          --ssh-flag="-o ServerAliveInterval=5" \
          --command="docker logs --follow ${{ env.ZEBRA_CONTAINER }}"
        env:
          ZEBRA_CONTAINER: ${{ steps.get-container-name.outputs.zebra_container }}

      - name: Delete test instance
        # Do not delete the instance if the sync timeouts in GitHub
        if: ${{ steps.full-sync-mainnet.outcome == 'success' || steps.full-sync-mainnet.outcome == 'failure' }}
        continue-on-error: true
        run: |
          gcloud compute instances delete "sync-tests-${{ env.GITHUB_HEAD_REF_SLUG_URL || env.GITHUB_REF_SLUG_URL }}-${{ env.GITHUB_EVENT_PULL_REQUEST_HEAD_SHA_SHORT || env.GITHUB_SHA_SHORT }}" --delete-disks all --zone "${{ env.ZONE }}"
```

**Review by dconnolly — APPROVED** [review]
👍

**Pull request merged**